golang/cosmos/daemon/main.go

```diff
@@ -55,6 +57,14 @@ func Run() {
 
 // RunWithController starts the app with a custom upcall handler.
 func RunWithController(sendToController cmd.Sender) {
+	// Exit on several different signals.
```

Comment:
You're probably right that SIGQUIT interception is not necessary.

The other two are needed because when I run `ag-chain-cosmos`, hitting Ctrl-C or `kill -TERM <pid>` don't interrupt the Cosmos layer.  This is probably because it's loaded as a shared library, so is not installing its usual signal handlers.